nuxtjs3/components/TVChartContainer.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <div :id="containerId" class="TVChartContainer" />
+</template>
+
+<script>
+
+function getLanguageFromURL() {
+  // eslint-disable-next-line prefer-regex-literals
+  const regex = new RegExp('[\\?&]lang=([^&#]*)')
+  const results = regex.exec(window.location.search)
+  return results === null ? null : decodeURIComponent(results[1].replace(/\+/g, ' '))
+}
+
+export default {
+  name: 'TVChartContainer',
+  props: {
+    symbol: {
+      default: 'AAPL',
+      type: String
+    },
+    interval: {
+      default: 'D',
+      type: String
+    },
+    containerId: {
+      default: 'tv_chart_container',
+      type: String
+    },
+    datafeedUrl: {
+      default: 'https://demo_feed.tradingview.com',
+      type: String
+    },
+    libraryPath: {
+      default: '/charting_library/',
+      type: String
+    },
+    chartsStorageUrl: {
+      default: 'https://saveload.tradingview.com',
+      type: String
+    },
+    chartsStorageApiVersion: {
+      default: '1.1',
+      type: String
+    },
+    clientId: {
+      default: 'tradingview.com',
+      type: String
+    },
+    userId: {
+      default: 'public_user_id',
+      type: String
+    },
+    fullscreen: {
+      default: false,
+      type: Boolean
+    },
+    autosize: {
+      default: true,
+      type: Boolean
+    },
+    studiesOverrides: {
+      type: Object,
+      default: () => ({})
+    }
+  },
+
+  tvWidget: null,
+
+  mounted() {
+    const widgetOptions = {
+      symbol: this.symbol,
+      // BEWARE: no trailing slash is expected in feed URL
+      datafeed: new window.Datafeeds.UDFCompatibleDatafeed(this.datafeedUrl),
+      interval: this.interval,
+      container: this.containerId,
+      library_path: this.libraryPath,
+
+      locale: getLanguageFromURL() || 'en',
+      disabled_features: ['use_localstorage_for_settings'],
+      enabled_features: ['study_templates'],
+      charts_storage_url: this.chartsStorageUrl,
+      charts_storage_api_version: this.chartsStorageApiVersion,
+      client_id: this.clientId,
+      user_id: this.userId,
```

Comment:
Maybe we should remove these?
By default vite will open the dev preview with `http://127.0.0.1:5173` and our save-load server will block the requests. However, our server is happy with `http://localhost:5173` because `localhost` has been whitelisted.

Alternatively, we can do this for the `vite.config.js` file:
```js
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import dns from 'dns'

dns.setDefaultResultOrder('verbatim');

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [vue()],
})
```

Might be worth doing this for the other examples using Vite.

Reply:
I've lazily copied/pasted from the other nuxtjs example.

Locally I always get localhost as a default & primary url.
Personally I won't bother adding an extra config/plugin. Guess if ppl reach that point they would be far ahead in their project and should be able to address that on their own.